examples/siro_sandbox/controllers/controller_helper.py

```diff
@@ -97,23 +98,39 @@ def __init__(
                     gui_input=gui_input,
                 )
             self.controllers.append(gui_agent_controller)
-
             if is_multi_agent:
-                self.controllers.append(
-                    SingleAgentBaselinesController(
-                        0 if self._gui_controlled_agent_index == 1 else 1,
-                        is_multi_agent,
-                        config,
-                        self._gym_habitat_env,
+                if args.app_state == "fetch":
+                    self.controllers.append(
+                        FetchBaselinesController(
+                            0 if self._gui_controlled_agent_index == 1 else 1,
+                            is_multi_agent,
+                            config,
+                            self._gym_habitat_env,
+                        )
+                    )
+                else:
+                    self.controllers.append(
+                        SingleAgentBaselinesController(
+                            0 if self._gui_controlled_agent_index == 1 else 1,
+                            is_multi_agent,
+                            config,
+                            self._gym_habitat_env,
+                        )
                     )
-                )
 
     def get_gui_agent_controller(self) -> Optional[Controller]:
         if self._gui_controlled_agent_index is None:
             return None
 
         return self.controllers[0]
 
+    def get_state_machine_agent_controller(self) -> Optional[Controller]:
+        if self._gui_controlled_agent_index is None:
+            return None
+
+        # This is pretty hacky
+        return self.controllers[1]
```

Comment:
Let's fix this magic number sloppiness. I put a comment in `get_gui_agent_controller` that covers this, too.